src/tasks/deploy-sync.js

```diff
@@ -28,10 +29,16 @@ gulp.task('deploy:sync-init', () => {
   const environment = config.environment.split(/\s*,\s*|\s+/)[0];
 
   const envObj = tkConfig[environment];
+  const valid_id = utils.validateThemeId(envObj.theme_id);
   let proxyTarget = `https://${envObj.store}`;
 
-  if (envObj.theme_id && (envObj.theme_id === parseInt(envObj.theme_id, 10))) {
-    proxyTarget += `?preview_theme_id=${envObj.theme_id}`;
+  if (!valid_id) {
+    messages.invalidThemeId(envObj.theme_id, environment);
+    process.exit();
+  }
+
+  if (valid_id && typeof(valid_id) === "number") {
```

Comment:
if you had a number as a string here (`valid_id = "1234"`), this wouldn't append the ID to the `proxyTarget`. I think if `valid_id` doesn't equal live would be good.